arm/Microsoft.OperationalInsights/workspaces/deploy.bicep

```diff
@@ -0,0 +1,844 @@
+  name: 'sampleLinuxPerf1'
+  kind: 'LinuxPerformanceObject'
+  properties: {
+    performanceCounters: [
+      {
+        counterName: '% Used Inodes'
+      }
+      {
+        counterName: 'Free Megabytes'
+      }
+      {
+        counterName: '% Used Space'
+      }
+      {
+        counterName: 'Disk Transfers/sec'
+      }
+      {
+        counterName: 'Disk Reads/sec'
+      }
+      {
+        counterName: 'Disk Writes/sec'
+      }
+    ]
+    objectName: 'Logical Disk'
+    instanceName: '*'
+    intervalSeconds: 10
+  }
+}
+
+resource logAnalyticsWorkspace_datasources_sampleLinuxPerfCollection1 'datasources@2020-03-01-preview' = {
+  name: 'sampleLinuxPerfCollection1'
+  kind: 'LinuxPerformanceCollection'
+  properties: {
+    state: 'Enabled'
+  }
+}
+
+resource logAnalyticsWorkspace_datasources_AzureActivityLog 'datasources@2020-03-01-preview' = [for item in activityLogAdditionalSubscriptionIDs: {
+  kind: 'AzureActivityLog'
+  name: '${(empty(activityLogAdditionalSubscriptionIDs) ? 'placeholder' : item)}'
+  properties: {
+    linkedResourceId: '/subscriptions/${item}/providers/microsoft.insights/eventTypes/management'
+  }
+  dependsOn: [
+    logAnalyticsWorkspace
+  ]
+}]
+
+resource storageinsightconfigs 'storageinsightconfigs@2020-03-01-preview' = if (!empty(diagnosticStorageAccountId)) {
+  name: '${diagnosticStorageAccountName}'
+  properties: {
+    containers: []
+    tables: [
+      'WADWindowsEventLogsTable'
+      'WADETWEventTable'
+      'WADServiceFabric*EventTable'
+      'LinuxsyslogVer2v0'
+    ]
+    storageAccount: {
+      id: diagnosticStorageAccountId
+      key: (empty(diagnosticStorageAccountId) ? '' : listKeys(diagnosticStorageAccountId, '2016-12-01').keys[0].value)
+    }
+  }
+}
+
+resource linkedServices 'linkedServices@2020-03-01-preview' = if (!empty(automationAccountId)) {
+  name: 'Automation'
+  properties: {
+    resourceId: automationAccountId
+  }
+}
+}
+
+@batchSize(1) // Serial loop deployment
+resource solutionRes 'Microsoft.OperationsManagement/solutions@2015-11-01-preview' = [for solution in solutions: if (!empty(solutions)) {
+  name: (empty(solutions) ? 'dummy' : '${solution}(${logAnalyticsWorkspaceName})')
+  location: location
+  properties: {
+    workspaceResourceId: logAnalyticsWorkspace.id
+  }
+  plan: {
+    name: (empty(solutions) ? 'dummy' : '${solution}(${logAnalyticsWorkspaceName})')
+    product: (empty(solutions) ? 'dummy' : 'OMSGallery/${solution}')
+    promotionCode: ''
+    publisher: 'Microsoft'
+  }
+  dependsOn: [
+    logAnalyticsWorkspace
+  ]
+}]
+
+resource logAnalyticsWorkspace_lock 'Microsoft.Authorization/locks@2016-09-01' = if (lockForDeletion) {
+  name: '${logAnalyticsWorkspaceName}-LADoNotDelete'
+  properties: {
+    level: 'CanNotDelete'
+  }
+  scope: logAnalyticsWorkspace
+}
+
+module rbac_name './.bicep/nested_rbac.bicep' = [for (item, i) in roleAssignments: {
```

Comment:
I'd suggest `logAnalyticsWorkspace_rbac`

Reply:
I'd suggest `[for (roleAssignment, index) in roleAssignments: {`

Reply:
done